jobs/haproxy/templates/haproxy.config.erb

```diff
@@ -622,22 +665,29 @@ backend tcp-<%= tcp_proxy["name"] %>
     resolvers = "resolvers default "
   end
   backend_ssl = ""
-  if tcp_proxy["backend_ssl"] then
-    if tcp_proxy["backend_ssl"].downcase == "verify" then
+
+  if (tcp_proxy["backend_ssl"] || "").downcase != "verify"
+    if tcp_proxy["backend_verifyhost"]
+      abort "Conflicting configuration: backend_ssl must be 'verify' to use backend_verifyhost in tcp backend configuration"
+    end
+  end
+
+  if tcp_proxy["backend_ssl"]
```

Comment:
same here, we could wrap this into one `if` with line 669

Reply:
This also suffers from the `elsif` on 681 meaning we'd get triple-nested ifs

The current approach does have some redundancy but it lowers the number of nested conditionals (which this file seems to have a lot of) and keeps the error checking separate

Reply:
ok let's keep it at that for now. maybe we can rework this altogether later and e.g. pull out the entire error checking into a separate section later so that we get rid of all the ifs and elses here.